tests/test_packages.py

```diff
@@ -0,0 +1,38 @@
+import json
+
+from cf_remote.packages import Release
+
+
+def test_release():
+    with open("tests/master-releases.json") as file:
+        data = file.read()
+    json_data = json.loads(data)
+    data = {
+        "URL": "test-release-url",
+        "version": "master",
+        "lts_branch": "test-lts-branch",
+        "data": json_data,
+    }
+    release = Release(data)
+    release.init_download()
+    found = release.find(["arm64", "ubuntu", "20"])
+    assert (
+        found[0].filename
+        == "cfengine-nova-hub_3.21.0a.138df3742~21749.ubuntu22_arm64.deb"
+    )
```

Comment:
So the code is supposed to find Ubuntu 22 packages for tags `["arm64", "ubuntu", "20"]`? That seems wrong. Also, are you sure that the order of things in `found` is guaranteed and consistent?

Reply:
No. For `["arm64", "ubuntu", "20"]` I would expect it to return the debian11_arm64 build since there is no ubuntu20 build in my test data (or in our real live data).

As for the order, I will check.

I also wonder if more than one is `found` which are used to install. I assumed it was the first, `[0]` but will check.

Reply:
It would seem that `Release.init_download()` uses the JSON data and processes that "in order". I suppose if `get_json()` doesn't always return in the order from the file things could change. And certainly if I was relying on a URL/live resource things could change, but in this case the data is local and static and paired with the test so I think that aspect is OK.

Reply:
`remote.py` uses the result from `find()` and takes `[-1]` so the last one found. I don't think this affects this test as I am asserting that two results are found and valid.

Reply:
I would say your example of `"ubuntu", "20"` is not appropriate here at this time because `get_package_tags()` will add something more like `ubuntu20` which will work properly to select `debian11`.

https://github.com/cfengine/cf-remote/blob/master/cf_remote/remote.py#L124-L132

Reply:
Indeed the order seems unstable, only in python 3.5: 

```
>       assert (
            found[0].filename
            == "cfengine-nova-hub_3.21.0a.138df3742~21749.debian11_arm64.deb"
        )
E       AssertionError: assert 'cfengine-nov...n11_arm64.deb' == 'cfengine-nov...n11_arm64.deb'
E         - cfengine-nova-hub_3.21.0a.138df3742~21749.debian11_arm64.deb
E         ?              ----
E         + cfengine-nova_3.21.0a.138df[37](https://github.com/cfengine/cf-remote/runs/7842247316?check_suite_focus=true#step:6:38)42~21749.debian11_arm64.deb
```

So I will fix that! :)

Reply:
Actually, this brings up an interesting detail: cf-remote with python 3.5 will likely install a different package right? Because the order of the result of `find()` is different so `install_host()` will use `[-1]` of that list: https://github.com/cfengine/cf-remote/blob/master/cf_remote/remote.py#L342

Reply:
Yes, that's a bug.

Reply:
For this it would seem the crux is whether `requests.json()` returns different order or not. I was loading a file and using `json.loads()` and in python 3.5 that didn't preserve the order of the file contents.

I will test 3.5.

Reply:
Yes, it seems that `response.json()` returns different order on 3.5 from 3.9

3.5:

```
Available CFEngine versions:
master, 3.20.0, 3.18.x, 3.18.2, 3.18.1, 3.18.0, 3.15.x, 3.15.6, 3.15.5, 3.15.4, 3.15.3, 3.15.2, 3.15.1, 3.15.0, 3.15.0b1
master artifacts are [cfengine-masterfiles-3.21.0a.dac58df60-1.pkg.tar.gz (any no_arch gz masterfiles cfengine 0a dac58df60 pkg tar client agent additional_assets), cfengine-nova-3.21.0a.59a4e502a-22153.x86_64.pkg.tar.gz (any x86_64 gz nova 64 cfengine 0a 59a4e502a pkg tar client agent additional_assets),
```

3.9:

```
Available CFEngine versions:
master, 3.20.0, 3.18.x, 3.18.2, 3.18.1, 3.18.0, 3.15.x, 3.15.6, 3.15.5, 3.15.4, 3.15.3, 3.15.2, 3.15.1, 3.15.0, 3.15.0b1
master artifacts are [cfengine-nova-hub_3.21.0a.59a4e502a~22159.debian9_amd64.deb (any x86_64 deb debian nova 64 cfengine hub 0a 59a4e502a22159 debian9 amd64 policy_server ubuntu17 ubuntu16 ubuntu), cfengine-nova-hub_3.21.0a.59a4e502a~22159.debian10_amd64.deb (any x86_64 deb debian nova 64 cfengine hub 0a 59a4e502a22159 debian10 amd64 policy_server ubuntu19 ubuntu18 ubuntu),
```

I think I have a fix. Will test and squash commits if it works.

Reply:
Ok. I made a separate commit to address this ordering problem with python 3.5 versus >3.5.